src/lib/transport.js

```diff
@@ -0,0 +1,260 @@
+
+  if (!transfer.executionCondition) {
+    debug('notified of transfer without executionCondition ', transfer)
+    return yield _reject(plugin, transfer.id, {
+      code: 'S00',
+      name: 'Bad Request',
+      message: 'got notification of transfer without executionCondition'
+    })
+  }
+
+  if (!transfer.ilp && !transfer.data) {
+    debug('got notification of transfer with no packet attached')
+    return yield _reject(plugin, transfer.id, {
+      code: 'S01',
+      name: 'Invalid Packet',
+      message: 'got notification of transfer with no packet attached'
+    })
+  }
+
+  const parsed = Packet.parseFromTransfer(transfer)
+  if (parsed === undefined) {
+    return yield _reject(plugin, transfer.id, {
+      code: 'S01',
+      name: 'Invalid Packet',
+      message: 'got notification of transfer with invalid ILP packet'
+    })
+  }
+
+  const destinationAmount = parsed.amount
+  const destinationAccount = parsed.account
+  const data = parsed.data
+
+  if (destinationAccount.indexOf(account) !== 0) {
+    debug('notified of transfer for another account: account=' +
+      destinationAccount +
+      ' me=' +
+      account)
+    return 'not-my-packet'
+  }
+
+  const localPart = destinationAccount.slice(account.length + 1)
+  const [ addressProtocol, addressReceiverId ] = localPart.split('.')
+
+  if (addressProtocol !== '~' + protocol) {
+    debug('notified of transfer with protocol=' + addressProtocol)
+    return 'not-my-packet'
+  }
+
+  if (addressReceiverId !== receiverId) {
+    debug('notified of transfer for another receiver: receiver=' +
+      addressReceiverId +
+      ' me=' +
+      receiverId)
+    return 'not-my-packet'
+  }
+
+  const decryptedData = _safeDecrypt(data, secret)
+
+  if (decryptedData === undefined) {
+    return yield _reject(plugin, transfer.id, {
+      code: 'S01',
+      name: 'Invalid Packet',
+      message: 'got notification of packet with corrupted ciphertext'
+    })
+  }
+
+  const expiresAt = decryptedData.expires_at
+  const amount = new BigNumber(transfer.amount)
+
+  if (amount.lessThan(destinationAmount)) {
+    debug('notified of transfer amount smaller than packet amount:' +
+      ' transfer=' + transfer.amount +
+      ' packet=' + destinationAmount)
+    return yield _reject(plugin, transfer.id, {
+      code: 'S04',
+      name: 'Insufficient Destination Amount',
+      message: 'got notification of transfer where amount is less than expected'
+    })
+  }
+
+  if (!allowOverPayment && amount.greaterThan(destinationAmount)) {
+    debug('notified of transfer amount larger than packet amount:' +
+      ' transfer=' + transfer.amount +
+      ' packet=' + destinationAmount)
+    return yield _reject(plugin, transfer.id, {
+      code: 'S03',
+      name: 'Invalid Amount',
+      message: 'got notification of transfer where amount is more than expected'
+    })
+  }
+
+  if (expiresAt && moment().isAfter(expiresAt)) {
```

Comment:
Since we're not really using much of `moment`'s advanced functionality I think it might be worth removing that dependency (doesn't have to be done in this PR though)